src/Pulsar.Client/Schema/AvroSchema.fs

```diff
@@ -6,22 +6,38 @@ open Avro.Generic
 open Avro.IO
 open Avro.Reflect
 open System.IO
+open Avro.Specific
 open Pulsar.Client.Api
 open AvroSchemaGenerator
 open Pulsar.Client.Common
 
-type internal AvroSchema<'T> private (stringSchema) =
+type internal AvroSchema<'T> private (schema: Schema, avroReader: DatumReader<'T>, avroWriter: DatumWriter<'T>) =
     inherit ISchema<'T>()
     let parameterIsClass =  typeof<'T>.IsClass
-    let avroSchema = Schema.Parse(stringSchema)
-    let avroWriter = ReflectWriter<'T>(avroSchema)
-    let avroReader = ReflectReader<'T>(avroSchema, avroSchema)
     
-    new () = AvroSchema(typeof<'T>.GetSchema())
+    new () =
+         let tpe = typeof<'T>
+         if typeof<ISpecificRecord>.IsAssignableFrom(tpe) then
+            let avroSchema = downcast tpe.GetField("_SCHEMA").GetValue(null) :Schema
+            let avroWriter = SpecificDatumWriter<'T>(avroSchema)
+            // note afaik for schema evolutions, the writerSchema (first param) should come from the topic
+            // (i.e the schema the value was written in) 
```

Comment:
Right! Therefore can you please update GetSpecificSchema, since it takes the schema from the topic as the first parameter

Reply:
ah, okay, i was wondering about that

Reply:
something like this:
```f#
    override this.GetSpecificSchema stringSchema =
        let writtenSchema = Schema.Parse(stringSchema)
        if avroReader :? SpecificDatumReader<'T> then
            AvroSchema(schema, SpecificDatumReader(writtenSchema, schema), avroWriter) :> ISchema<'T>
        else
            AvroSchema(stringSchema) :> ISchema<'T>
```

Reply:
looks good to me, the only thing to change is to put the parse logic inside if rather than outside

Reply:
though wouldn't the reflection based version also need the same thing?
```f#
    override this.GetSpecificSchema stringSchema =
        let writtenSchema = Schema.Parse(stringSchema)
        if avroReader :? SpecificDatumReader<'T> then
            AvroSchema(schema, SpecificDatumReader(writtenSchema, schema), avroWriter) :> ISchema<'T>
        else
            AvroSchema(schema, ReflectReader<'T>(writtenSchema, schema), avroWriter) :> ISchema<'T>
```

Reply:
Hm... Good catch! I think you are right